From 3ed98f00a5294193a85f59f6b4a828292f53cab9 Mon Sep 17 00:00:00 2001
From: John Brawn <john.brawn@arm.com>
Date: Thu, 3 Nov 2016 13:55:04 +0000
Subject: [PATCH 6/8] [CMake] Make CMAKE_INSTALL_RPATH work again

r285714 made it so that when CMAKE_INSTALL_RPATH is set _install_rpath is not
set, but that means INSTALL_RPATH gets set to an empty string which isn't what
we want. Fix this by setting INSTALL_RPATH only when _install_rpath is set.

git-svn-id: https://llvm.org/svn/llvm-project/llvm/trunk@285908 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit a0e0673b0f73cebabf73e8a7817e871ad54cb24f)
---
 cmake/modules/AddLLVM.cmake | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git llvm_release_39/cmake/modules/AddLLVM.cmake macports_release_39/cmake/modules/AddLLVM.cmake
index 3b42ee53050..bb614c3a7bc 100644
--- llvm_release_39/cmake/modules/AddLLVM.cmake
+++ macports_release_39/cmake/modules/AddLLVM.cmake
@@ -1323,8 +1323,10 @@ function(llvm_setup_rpath name)
     return()
   endif()
 
-  set_target_properties(${name} PROPERTIES
-                        BUILD_WITH_INSTALL_RPATH On
-                        INSTALL_RPATH "${_install_rpath}"
-                        ${_install_name_dir})
+  if(DEFINED _install_rpath)
+    set_target_properties(${name} PROPERTIES
+                          BUILD_WITH_INSTALL_RPATH On
+                          INSTALL_RPATH "${_install_rpath}"
+                          ${_install_name_dir})
+  endif()
 endfunction()
-- 
2.11.0

